### PR TITLE
Meta+Everywhere: Enforce Markdown formatting

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This was supposed to be a simple "enable Markdown formatting in our prettier script" commit, but turned out to require extra support from our Markdown parsing library and a fix to the ports table regex. Either way, the +-3000 is just auto-formatting and doesn't need to be reviewed; it's in the separate commit 4.

### LibMarkdown: Support escaped links

Prettier automatically adds <> escapes around
invalid characters in links, as allowed by
CommonMark:
<https://spec.commonmark.org/0.31.2/#link-destination>.
This will come up in CatDog.md, where a Wikipedia
link uses ambiguous parentheses that must be
escaped like this. This commit adds support for
parsing these kinds of links. There is no
indication in the syntax tree on what kind of link
it originally was since this information doesn't
seem relevant to renderers.

### Meta: Ignore auto-generated and incompatible files in prettier

Several auto-generated Markdown files should be
excluded once we check Markdown. One JS regex test
isn't compatible with prettier's parser.

### Meta: Fix Ports table linter when there's only one space after the link

With prettier formatting, at least one port's row
will be formatted such that there is only one
space, but the previous regex required at least
two spaces. Disregarding the fact that we probably
shouldn't parse markdown with a regex, this fixes
the issue for now.

### Everywhere: Format Markdown with prettier

These are the exact changes necessary to get the
next commit to pass pre-commit checks.

### Meta: Enforce Prettier formatting of Markdown files

We already use the opinionated Prettier formatter
for JavaScript (with appropriate configuration),
and in general it seems to be one of the best
Markdown auto-formatters. Therefore, the choice
seems natural to finally enforce Markdown
formatting.

### CODE REVIEW

File: Meta/lint-prettier.sh [hunk 1]
```diff
@@ -10,20 +10,20 @@ if [ "$#" -eq "0" ]; then
...
             --exclude-from .prettierignore \
             -- \
-            '*.js' '*.mjs'
+            '*.js' '*.mjs' '*.md'
     )
 else
     files=()
     for file in "$@"; do
-        if [[ "${file}" == *".js" ]] || [[ "${file}" == *".mjs" ]]; then
+        if [[ "${file}" == *".js" ]] || [[ "${file}" == *".mjs" ]] || [[ "${file}" == *".md" ]]; then
```
Comment (sdomi) [inline Meta/lint-prettier.sh:18]: nit: can we just `[[ "${file}" == *".js" || "${file}" == *".mjs" || "${file}" == *".md" ]]`? either one is equal but the current version hurts to look at more

**Comment by sdomi** [comment]
this is very cool!! good job, i'm in full support <3

**Comment by supercomputer7** [comment]
Seems like a decent change to me, thank you for improving markdown correctness :)

File: .prettierignore [hunk 2]
```diff
@@ -1,7 +1,13 @@
+# Automatically regenerated Markdown files based on git patches.
+Toolchain/Patches
+Ports/*/patches
+
 Base/home/anon/Source/js
 Userland/Libraries/LibJS/Tests/invalid-lhs-in-assignment.js
 Userland/Libraries/LibJS/Tests/unicode-identifier-escape.js
 Userland/Libraries/LibJS/Tests/modules/failing.mjs
+# FIXME: Uses nonstandard regex flags which cause a syntax error in prettier.
```
Comment (alimpfard) [inline .prettierignore:9]: Just slap a `// prettier-ignore` on line 8.
`v` _is_ a standard flag (https://tc39.es/ecma262/#sec-get-regexp.prototype.unicodesets)

Comment (kleinesfilmroellchen) [reply .prettierignore:9]: I'm surprised prettier is so bad at this

Comment (kleinesfilmroellchen) [reply .prettierignore:9]: This doesn't work, since prettier fails to compile the file at all with this flag present. I'll change the comment though.

**Pull request merged**